src/plugins/data_views/common/data_views/abstract_data_views.ts

```diff
@@ -442,6 +442,26 @@ export abstract class AbstractDataView {
     return this.upsertScriptedFieldInternal(field);
   }
 
+  /**
+   * Only used by search source to process sorting of scripted fields
+   * @param name field name
+   * @returns DataViewFieldBase
+   */
+  getIfScriptedField(name: string): DataViewFieldBase | undefined {
```

Comment:
One nit about naming, maybe for consistency, there's a function called `hasRuntimeField` in this file, so I think `hasScriptedField`, at least for me it's clear on the first sight what it's doing. Apart from that, I really like this improvement, so it's one workaround less necessary to omit field list loading

Reply:
I propose `getScriptedField` as this method is returning the field.
Also the code looks pretty much similar to `getRuntimeField`